src/main/java/ru/leo/models/model/ModelServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package ru.leo.models.model;
+
+import com.google.inject.Inject;
+import java.util.Collection;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ModelServiceImpl implements ModelService {
+    private final Map<String, Model> models = new ConcurrentHashMap<>();
```

Comment:
```suggestion
    private final ConcurrentMap<String, Model> models = new ConcurrentHashMap<>();
```

Дабы явно показывать намерение